src/hb-aat.h

```diff
@@ -0,0 +1,765 @@
+{
+  HB_AAT_LAYOUT_FEATURE_TYPE_ALL_TYPOGRAPHIC                   = 0,
+  HB_AAT_LAYOUT_FEATURE_TYPE_LIGATURES                         = 1,
+  HB_AAT_LAYOUT_FEATURE_TYPE_CURISVE_CONNECTION                = 2,
+  HB_AAT_LAYOUT_FEATURE_TYPE_LETTER_CASE                       = 3,
+  HB_AAT_LAYOUT_FEATURE_TYPE_VERTICAL_SUBSTITUTION             = 4,
+  HB_AAT_LAYOUT_FEATURE_TYPE_LINGUISTIC_REARRANGEMENT          = 5,
+  HB_AAT_LAYOUT_FEATURE_TYPE_NUMBER_SPACING                    = 6,
+  HB_AAT_LAYOUT_FEATURE_TYPE_SMART_SWASH_TYPE                  = 8,
+  HB_AAT_LAYOUT_FEATURE_TYPE_DIACRITICS_TYPE                   = 9,
+  HB_AAT_LAYOUT_FEATURE_TYPE_VERTICAL_POSITION                 = 10,
+  HB_AAT_LAYOUT_FEATURE_TYPE_FRACTIONS                         = 11,
+  HB_AAT_LAYOUT_FEATURE_TYPE_OVERLAPPING_CHARACTERS_TYPE       = 13,
+  HB_AAT_LAYOUT_FEATURE_TYPE_TYPOGRAPHIC_EXTRAS                = 14,
+  HB_AAT_LAYOUT_FEATURE_TYPE_MATHEMATICAL_EXTRAS               = 15,
+  HB_AAT_LAYOUT_FEATURE_TYPE_ORNAMENT_SETS_TYPE                = 16,
+  HB_AAT_LAYOUT_FEATURE_TYPE_CHARACTER_ALTERNATIVES            = 17,
+  HB_AAT_LAYOUT_FEATURE_TYPE_DESIGN_COMPLEXITY_TYPE            = 18,
+  HB_AAT_LAYOUT_FEATURE_TYPE_STYLE_OPTIONS                     = 19,
+  HB_AAT_LAYOUT_FEATURE_TYPE_CHARACTER_SHAPE                   = 20,
+  HB_AAT_LAYOUT_FEATURE_TYPE_NUMBER_CASE                       = 21,
+  HB_AAT_LAYOUT_FEATURE_TYPE_TEXT_SPACING                      = 22,
+  HB_AAT_LAYOUT_FEATURE_TYPE_TRANSLITERATION                   = 23,
+  HB_AAT_LAYOUT_FEATURE_TYPE_ANNOTATION_TYPE                   = 24,
+  HB_AAT_LAYOUT_FEATURE_TYPE_KANA_SPACING_TYPE                 = 25,
+  HB_AAT_LAYOUT_FEATURE_TYPE_IDEOGRAPHIC_SPACING_TYPE          = 26,
+  HB_AAT_LAYOUT_FEATURE_TYPE_UNICODE_DECOMPOSITION_TYPE        = 27,
+  HB_AAT_LAYOUT_FEATURE_TYPE_RUBY_KANA                         = 28,
+  HB_AAT_LAYOUT_FEATURE_TYPE_CJK_SYMBOL_ALTERNATIVES_TYPE      = 29,
+  HB_AAT_LAYOUT_FEATURE_TYPE_IDEOGRAPHIC_ALTERNATIVES_TYPE     = 30,
+  HB_AAT_LAYOUT_FEATURE_TYPE_CJK_VERTICAL_ROMAN_PLACEMENT_TYPE = 31,
+  HB_AAT_LAYOUT_FEATURE_TYPE_ITALIC_CJK_ROMAN                  = 32,
+  HB_AAT_LAYOUT_FEATURE_TYPE_CASE_SENSITIVE_LAYOUT             = 33,
+  HB_AAT_LAYOUT_FEATURE_TYPE_ALTERNATE_KANA                    = 34,
+  HB_AAT_LAYOUT_FEATURE_TYPE_STYLISTIC_ALTERNATIVES            = 35,
+  HB_AAT_LAYOUT_FEATURE_TYPE_CONTEXTUAL_ALTERNATIVES           = 36,
+  HB_AAT_LAYOUT_FEATURE_TYPE_LOWER_CASE                        = 37,
+  HB_AAT_LAYOUT_FEATURE_TYPE_UPPER_CASE                        = 38,
+  HB_AAT_LAYOUT_FEATURE_TYPE_LANGUAGE_TAG_TYPE                 = 39,
+  HB_AAT_LAYOUT_FEATURE_TYPE_CJK_ROMAN_SPACING_TYPE            = 103,
+  HB_AAT_LAYOUT_FEATURE_TYPE_UNDEFINED                         = 0xFFFF
+} hb_aat_layout_feature_type_t;
+
+
+/*
+ * Selectors for HB_AAT_LAYOUT_FEATURE_TYPE_ALL_TYPOGRAPHIC
+ *
+ * Since: REPLACEME
+ */
+enum
+{
+  HB_AAT_LAYOUT_SELECTOR_ALL_TYPE_FEATURES_ON  = 0,
```

Comment:
I don't like that the name of these selectors doesn't reflect what feature type they apply on.  But guess that's life.